src/AzureRenderManager/WebApp/CostManagement/CostManagementClient.cs

```diff
@@ -0,0 +1,284 @@
+
+            public new IReadOnlyList<FilterExpression> And { get; }
+        }
+
+        private class OrExpression : FilterExpression
+        {
+            public OrExpression(params FilterExpression[] or)
+            {
+                Or = or.ToArray();
+            }
+
+            public new IReadOnlyList<FilterExpression> Or { get; }
+        }
+
+        private class TagExpression : FilterExpression
+        {
+            public TagExpression(OperatorExpression expression)
+                => Tags = expression;
+
+            public OperatorExpression Tags { get; }
+        }
+
+        private class DimensionExpression : FilterExpression
+        {
+            public DimensionExpression(OperatorExpression expression)
+                => Dimension = expression;
+
+            public new OperatorExpression Dimension { get; }
+        }
+
+        private class OperatorExpression
+        {
+            public OperatorExpression(string name, Operator op, IEnumerable<string> values)
+            {
+                Name = name;
+                Operator = op;
+                Values = values.ToList();
+            }
+
+            public string Name { get; }
+
+            [JsonConverter(typeof(StringEnumConverter))]
+            public Operator Operator { get; }
+
+            public IReadOnlyList<string> Values { get; }
+        }
+    }
+
+    public enum Timeframe
+    {
+        MonthToDate,
+        YearToDate,
+        WeekToDate,
+    }
+
+    public enum Granularity
+    {
+        Daily
+    }
+
+    public enum Operator
+    {
+        Contains,
+        In
+    }
+
+    public class CostManagementClient 
+    {
+        private readonly HttpClient _client;
+        private readonly string _accessToken;
+
+        public CostManagementClient(HttpClient client, string accessToken)
+        {
+            _client = client;
+            _accessToken = accessToken;
+        }
+
+        private static Uri GetUri(string scope)
+            => new Uri($"https://management.azure.com/{scope}/providers/Microsoft.CostManagement/query?api-version=2019-01-01");
+
+        public async Task<UsageResponse> GetUsageForResourceGroup(Guid subscriptionId, string resourceGroupName, UsageRequest usageRequest)
+        {
+            var uri = GetUri($"/subscriptions/{subscriptionId}/resourcegroups/{resourceGroupName}");
+
+            var request =
+                new HttpRequestMessage(HttpMethod.Post, uri)
+                {
+                    Content = new StringContent(JsonConvert.SerializeObject(usageRequest), Encoding.UTF8, "application/json"),
+                };
+
+            request.Headers.Authorization = new AuthenticationHeaderValue("Bearer", _accessToken);
+
+            using (var response = await _client.SendAsync(request))
+            {
+                return await response.Content.ReadAsAsync<UsageResponse>();
+            }
+        }
+    }
+
+    public sealed class CostManagementClientAccessor : NeedsAccessToken
```

Comment:
Can we add this to the ManagementClientProvider?

Reply:
Yes but it works a bit differently to the others (you pass it its own `HttpClient`, for example). Having it on its own means the `HttpClient` can be configured specifically for it, with a retry policy specifically for the cost-management endpoint.

Reply:
Ok makes sense.